### PR TITLE
DATA-2327 - PFS updates

### PR DESCRIPTION
**Pull request by mjmeans**

### CODE REVIEW

File: data/pathfinder_pfs/paizo/adventure_path/curse_of_the_crimson_throne/pfs_players_guide/pfs_curse_of_the_crimson_throne_players_guide.pcc [hunk 1]
```diff
@@ -40,5 +40,5 @@ COPYRIGHT:PCGen dataset conversion for "Curse of the Crimson Throne Player's Gui
 # Equipment: all equipment on pages 12-13 (except hide shirt); Feats: Acadamae Graduate, Crossbow Mastery
 EQUIPMENT:@/pathfinder/paizo/adventure_path/curse_of_the_crimson_throne/players_guide/cotctpg_equip_arms_armor.lst|(EXCLUDE:Hide Shirt)
 WEAPONPROF:@/pathfinder/paizo/adventure_path/curse_of_the_crimson_throne/players_guide/cotctpg_profs_weapon.lst
-ABILITY:@/pathfinder/paizo/adventure_path/curse_of_the_crimson_throne/players_guide/cotctpg_feats.lst|(INCLUDE:CATEGORY=Special Ability,Acadamae Graduate)
+FEAT:@/pathfinder/paizo/adventure_path/curse_of_the_crimson_throne/players_guide/cotctpg_feats.lst|(INCLUDE:CATEGORY=Special Ability,Acadamae Graduate)
```
Comment (LegacyKing) [inline data/pathfinder_pfs/paizo/adventure_path/curse_of_the_crimson_throne/pfs_players_guide/pfs_curse_of_the_crimson_throne_players_guide.pcc]: Why was this changed back to feat?

File: data/pathfinder_pfs/paizo/adventure_path/legacy_of_fire/pfs_players_guide/pfs_legacy_of_fire_players_guide.pcc [hunk 2]
```diff
@@ -44,7 +44,7 @@ INFOTEXT:This dataset uses trademarks and/or copyrights owned by Paizo Inc., whi
 # Traits: all traits on pages 7-9 are allowed except Duskwalker Agent
 # (Notes: Uwaga Highlander applies only to the Stealth skill and Eyes and Ears of the City applies only to the Perception skill; no campaign traits are allowed.)
 
-ABILITY:@/pathfinder/paizo/adventure_path/legacy_of_fire/players_guide/lofpg_feats.lst|(INCLUDE:CATEGORY=Special Ability,Sandwalker)
+FEAT:@/pathfinder/paizo/adventure_path/legacy_of_fire/players_guide/lofpg_feats.lst|(INCLUDE:CATEGORY=Special Ability,Sandwalker)
 EQUIPMENT:@/pathfinder/paizo/adventure_path/legacy_of_fire/players_guide/lofpg_equip_general.lst
```
Comment (LegacyKing) [inline data/pathfinder_pfs/paizo/adventure_path/legacy_of_fire/pfs_players_guide/pfs_legacy_of_fire_players_guide.pcc]: Same as before this is going backwards...

File: data/pathfinder_pfs/paizo/campaign_setting/pfs_dark_markets/pfs_dm_feats.lst [hunk 3]
```diff
@@ -2,7 +2,7 @@
...
 # Original Entry by: Terry Milnes
 
-#Pesh Euphoria.MOD		VISIBLE:NO	TYPE:PFSNotLegal
-#Pesh Healing.MOD		VISIBLE:NO	TYPE:PFSNotLegal
-#Pesh Rejuvenation.MOD		VISIBLE:NO	TYPE:PFSNotLegal
-#Sahir-Afiyun.MOD		VISIBLE:NO	TYPE:PFSNotLegal
+#CATEGORY=FEAT|Pesh Euphoria.MOD		!PRECHARACTERTYPE:1,PC	TYPE:PFSNotLegal
+#CATEGORY=FEAT|Pesh Healing.MOD		!PRECHARACTERTYPE:1,PC	TYPE:PFSNotLegal
+#CATEGORY=FEAT|Pesh Rejuvenation.MOD		!PRECHARACTERTYPE:1,PC	TYPE:PFSNotLegal
+#CATEGORY=FEAT|Sahir-Afiyun.MOD		!PRECHARACTERTYPE:1,PC	TYPE:PFSNotLegal
```
Comment (LegacyKing) [inline data/pathfinder_pfs/paizo/campaign_setting/pfs_dark_markets/pfs_dm_feats.lst]: If these are commented out why are we even listing them?
Why not decomment them?

File: data/pathfinder_pfs/paizo/campaign_setting/pfs_inner_sea_world_guide/pfs_iswg_equip_arms_armor.lst [hunk 4]
```diff
@@ -21,10 +21,10 @@ Field Plate.MOD			TYPE:PFSLegal
...
 Klar (as Shield).MOD		TYPE:PFSLegal
-Pepperbox.MOD			TYPE:PFSLegal	PREABILITY:1,CATEGORY=FEAT,Gunsmithing
-Pistol.MOD				TYPE:PFSLegal	PREABILITY:1,CATEGORY=FEAT,Gunsmithing
-Musket.MOD				TYPE:PFSLegal	PREABILITY:1,CATEGORY=FEAT,Gunsmithing
-Cannon.MOD				TYPE:PFSLegal	PREABILITY:1,CATEGORY=FEAT,Gunsmithing
+Pepperbox.MOD			TYPE:PFSLegal	PREFEAT:1,Gunsmithing
+Pistol.MOD				TYPE:PFSLegal	PREFEAT:1,Gunsmithing
+Musket.MOD				TYPE:PFSLegal	PREFEAT:1,Gunsmithing
+Cannon.MOD				TYPE:PFSLegal	PREFEAT:1,Gunsmithing
 Black Powder (Dose).MOD		TYPE:PFSLegal
```
Comment (LegacyKing) [inline data/pathfinder_pfs/paizo/campaign_setting/pfs_inner_sea_world_guide/pfs_iswg_equip_arms_armor.lst]: Moving backwards, PREFEAT is being removed.

File: data/pathfinder_pfs/paizo/campaign_setting/pfs_inner_sea_world_guide/pfs_iswg_profs_weapon.lst [hunk 5]
```diff
@@ -13,7 +13,7 @@ Aldori Dueling Sword.MOD	TYPE:PFSLegal
 Sawtooth Sabre.MOD		TYPE:PFSLegal
 Urumi.MOD				TYPE:PFSLegal
 Bladed Scarf.MOD			TYPE:PFSLegal
-Firearms.MOD			TYPE:PFSLegal	PREABILITY:1,CATEGORY=FEAT,Gunsmithing
+Firearms.MOD			TYPE:PFSLegal	PREFEAT:1,Gunsmithing
 Pepperbox.MOD			TYPE:PFSLegal
```
Comment (LegacyKing) [inline data/pathfinder_pfs/paizo/campaign_setting/pfs_inner_sea_world_guide/pfs_iswg_profs_weapon.lst]: Same thing, this is a backwards change.

File: data/pathfinder_pfs/paizo/campaign_setting/pfs_seekers_of_secrets/pfs_sos_classes.lst [hunk 6]
```diff
@@ -10,7 +10,7 @@ CLASS:Pathfinder Savant.FORGET
 # Class Name		Hit Dice	Type			Class Type		Abbreviation	Max Level	Source Page		Define		Combat bonus										Save bonus																						Modify VAR
 CLASS:Pathfinder Savant	HD:6		TYPE:PC.Prestige	CLASSTYPE:PC	ABB:PaS		MAXLEVEL:7	SOURCEPAGE:p.60	DEFINE:PaSLVL|0	BONUS:COMBAT|BASEAB|classlevel("APPLIEDAS=NONEPIC")/2|TYPE=Base.REPLACE	BONUS:SAVE|BASE.Will|(classlevel("APPLIEDAS=NONEPIC")+1)/2	BONUS:SAVE|BASE.Fortitude,BASE.Reflex|(classlevel("APPLIEDAS=NONEPIC")+1)/3	BONUS:VAR|PaSLVL|CL
 # Class Name		Required Feat							Required Skill									Req. Spell Type
-CLASS:Pathfinder Savant	PREABILITY:1,CATEGORY=FEAT,Magical Aptitude	PREABILITY:1,CATEGORY=FEAT,SpellFocus	PRESKILL:3,Knowledge (Arcana)=5,Spellcraft=5,Use Magic Device=5	PRESPELLTYPE:1,Arcane=2,Divine=2
+CLASS:Pathfinder Savant	PREFEAT:1,Magical Aptitude	PREFEAT:1,SpellFocus	PRESKILL:3,Knowledge (Arcana)=5,Spellcraft=5,Use Magic Device=5	PRESPELLTYPE:1,Arcane=2,Divine=2
 # Class Name		Skill Pts/Lvl	Class Skill
```
Comment (LegacyKing) [inline data/pathfinder_pfs/paizo/campaign_setting/pfs_seekers_of_secrets/pfs_sos_classes.lst]: Same thing, FEAT being put back in...

File: data/pathfinder_pfs/paizo/player_companion/pfs_animal_archive/pfs_animal_archive.pcc [hunk 7]
```diff
@@ -35,13 +35,22 @@ COPYRIGHT:PCGen dataset conversion for "Pathfinder Player Companion: Animal Arch
...
+# Feats: all feats on pages 18-19 are legal except Familiar Spell;
+# Magic items: all magic items on pages 26-27 are legal except circlet of speaking and familiar metamagic rod;
+# Spells: all spells on pages 24-25 are legal;
+# Tricks: All tricks on pages 8-9 are legal.
 
-PRECAMPAIGN:1,INCLUDES=Animal Archive
-#PCC:../animal_archive.pcc
+# Data files
+EQUIPMENT:pfs_anar_equipment.lst|PRECAMPAIGN:1,INCLUDES=Animal Archive
+FEAT:pfs_anar_feats.lst|PRECAMPAIGN:1,INCLUDES=Animal Archive
```
Comment (LegacyKing) [inline data/pathfinder_pfs/paizo/player_companion/pfs_animal_archive/pfs_animal_archive.pcc]: FEAT used instead of Ability.

File: data/pathfinder_pfs/paizo/roleplaying_game/pfs_core_rulebook/pfs_cr_classes.lst [hunk 8]
```diff
@@ -37,12 +37,12 @@ CLASS:Wizard.MOD		!PREALIGN:CE,NE,LE
 
 # Class Name
 CLASS:Cleric.MOD
-1	ABILITY:FEAT|AUTOMATIC|Improved Unarmed Strike|PREDEITY:1,Irori
+1	AUTO:FEAT|Improved Unarmed Strike|PREDEITY:1,Irori
```
Comment (LegacyKing) [inline data/pathfinder_pfs/paizo/roleplaying_game/pfs_core_rulebook/pfs_cr_classes.lst]: going backwards here...

File: data/pathfinder_pfs/paizo/roleplaying_game/pfs_core_rulebook/pfs_cr_classes.lst [hunk 9]
```diff
@@ -37,12 +37,12 @@ CLASS:Wizard.MOD		!PREALIGN:CE,NE,LE
...
 # Class Name
 CLASS:Cleric.MOD
-1	ABILITY:FEAT|AUTOMATIC|Improved Unarmed Strike|PREDEITY:1,Irori
+1	AUTO:FEAT|Improved Unarmed Strike|PREDEITY:1,Irori
 
 # Class Name
 CLASS:Wizard.MOD
 ###Block: Level progression
-1	BONUS:ABILITYPOOL|Spell Focus Replacement|1	ABILITY:FEAT|AUTOMATIC|.CLEAR.Scribe Scroll
+1	BONUS:ABILITYPOOL|Spell Focus Replacement|1	AUTO:FEAT|.CLEAR.Scribe Scroll
```
Comment (LegacyKing) [inline data/pathfinder_pfs/paizo/roleplaying_game/pfs_core_rulebook/pfs_cr_classes.lst]: AUTO:FEAT is going away.

File: data/pathfinder_pfs/paizo/roleplaying_game/pfs_ultimate_combat/pfs_uc_profs_weapon.lst [hunk 10]
```diff
@@ -2,4 +2,4 @@
 
 
 # Weapon Name	Required Feat
-Firearms.MOD	PREABILITY:1,CATEGORY=FEAT,Gunsmithing
+Firearms.MOD	PREFEAT:1,Gunsmithing
```
Comment (LegacyKing) [inline data/pathfinder_pfs/paizo/roleplaying_game/pfs_ultimate_combat/pfs_uc_profs_weapon.lst]: PREFEAT is going away.

File: data/pathfinder_pfs/paizo/adventure_path/curse_of_the_crimson_throne/pfs_players_guide/pfs_curse_of_the_crimson_throne_players_guide.pcc (hunk 1 above)
Comment (mjmeans) [inline data/pathfinder_pfs/paizo/adventure_path/curse_of_the_crimson_throne/pfs_players_guide/pfs_curse_of_the_crimson_throne_players_guide.pcc]: It wasn’t changed back to feat. I started this project a week before FEAT was decided to be removed. And the line changed for other reasons.

**Comment by LegacyKing** [comment]
Long review. Please correct the FEAT changes introduced by this PR and when complete, let me know. Cheers,

File: data/pathfinder_pfs/paizo/campaign_setting/pfs_dark_markets/pfs_dm_feats.lst (hunk 3 above)
Comment (mjmeans) [inline data/pathfinder_pfs/paizo/campaign_setting/pfs_dark_markets/pfs_dm_feats.lst]: These are items that are listed in multiple sources. PFS rules refer to them in this source specifically, so they are listed.

**Comment by mjmeans** [comment]
Okay I changed all AUTO:FEAT and PREFEAT to the correct form. However there are still some REMOVE:FEAT tags. How do I do that as abilities?

**Comment by LegacyKing** [comment]
As Tom mentioned, REMOVE:FEAT never worked for AUTO/VFEAT items. That should be handled using the same triggers we do for Archetypes and such. There is an ACF category - Alternate Class Features. Simply use that. Here is an example:
1 <> ABILITY:FEAT|AUTOMATIC|Scribe Scroll|!PREABILITY:1,CATEGORY=ACF,TYPE.WizardScribeScroll

Then grant to the Wizard
ABILITY:ACF|AUTOMATIC|Wizard ~ Remove Scribe Scroll

Wizard ~ Remove Scribe Scroll <> CATEGORY:ACF <> TYPE:WizardScribeScroll

Cheers,

**Comment by mjmeans** [comment]
Are you sure the ACF mechanism won't cause archetypes applicability to be affected? The PFS replacement rules replace the basic class rule without affecting anything else. I.e. any archetype that would replace that ability, replaces the PFS altered version instead. So an archetype that replaced the Scribe Scroll feat should still be usable even though Scribe Scroll has been itself replaced with Spell Focus (or whatever).


Also, I'm trying to do this without touching the released, well tested, original source sets. What you are suggesting seems to require changing the original source. There really should (in the future) be a way to MOD any portion of an item without having to resort to modifying released source sets. A much more robust and flexible MOD system. But I guess that is just wishful thinking.






Sent from Surface





From: Andrew Maitland
Sent: ‎Monday‎, ‎January‎ ‎26‎, ‎2015 ‎1‎:‎20‎ ‎AM
To: 'PCGen/pcgen'
Cc: Mark Means





As Tom mentioned, REMOVE:FEAT never worked for AUTO/VFEAT items. That should be handled using the same triggers we do for Archetypes and such. There is an ACF category - Alternate Class Features. Simply use that. Here is an example:
1 <> ABILITY:FEAT|AUTOMATIC|Scribe Scroll|!PREABILITY:1,CATEGORY=ACF,TYPE.WizardScribeScroll

Then grant to the Wizard
ABILITY:ACF|AUTOMATIC|Wizard ~ Remove Scribe Scroll

Wizard ~ Remove Scribe Scroll <> CATEGORY:ACF <> TYPE:WizardScribeScroll

Cheers,

**Comment by LegacyKing** [comment]
@mjmeans ,

Please cut out the email portion from me, it makes the conversation log doubly long on the web interface.

(1) FEAT --> ABILITY 
I'm still showing the following left unfixed:
pfs_curse_of_the_crimson_throne_players_guide.pcc 
pfs_players_guide/pfs_legacy_of_fire_players_guide.pcc 

(2) I was giving a general how-to. In a perfect world, PCGen programming would allow for specific object Modification without fail. However, we don't have a perfect world and PCGen was not designed with such modification capabilities. In fact, the Modification came after it came to light that things would be modified in later resources. History lesson shows this insidious cycle of "Rule" object to break said "Rule". D&D and the d20 system is all about rules, and breaking them. 

In this case, no, we will not be "Modifying" the original source. PFS has the distinction of being a switch out x for y rule system. In your case, the ACF is not going to work if you have an Archetype system to content with. And I really, really don't want to rewrite how classes work in data, and I doubt Tom or James wish to rewrite the system to be a lot more MOD friendly at this juncture, so your specific solution is going to be more invasive:
(a) For the Wizard - In the PFS file, EXCLUDE the wizard from being loaded in the core class file
(b) Make a complete copy of the "working" original wizard in a pfs specific file
(c) Swap out the Class Feature of Scribe Scroll for whatever else goes in its stead

That is the most logical and specific solution to handle this case. Since it's a PFS rule change, the PFS system now handles it. We can't have perfect synchronization in all cases, which is why I wanted to have PFS as its own system. 

Anyways, please tidying up the remaining FEAT > ABILITY and I'll pull this in.

Cheers,

**Comment by mjmeans** [comment]
It’s in. I’ll wait to do the class fix for a separate PR with I go through the remaining TODO’s in the PFS sets.

**Comment by LegacyKing** [comment]
Hm, something is in conflict. Might be my latest patch with AA. I'll check it out.

**Comment by LegacyKing** [comment]
@mjmeans pull from my PFS branch on legacyking, and merge that patch. Then push to your remote and it'll fix this file conflict.

**Comment by mjmeans** [comment]
I have no idea how to do that from the github.com web interface.

**Comment by mjmeans** [comment]
Command line interface… how droll. I think it’s done.
